File: pegasusX/apps/supplier-app-ios/SupplierApp/Views/Exceptions/ExceptionsList.swift
import SwiftUI

private let resolvableKinds: Set<String> = ["CASH_DISCREPANCY", "CREDIT_NOTE_DRAFT", "CREDIT_FREEZE"]

struct ExceptionsList: View {
    let rows: [SupplierExceptionRow]
    var busyKey: String? = nil
    var onResolve: (SupplierExceptionRow) -> Void = { _ in }

    var body: some View {
        ResponsiveGridContentWrapper {
            ForEach(rows) { row in
                VStack(alignment: .leading, spacing: 6) {
                    Text(row.orderId).font(.headline)
                    Text("\(row.kind) · \(row.status)").font(.subheadline)
                    if let note = row.note, !note.isEmpty { Text(note).font(.caption) }
                    if let manifestId = row.manifestId { Text("Manifest \(manifestId)").font(.caption) }
                    if resolvableKinds.contains(row.kind.uppercased()) {
                        let key = "\(row.kind):\(row.orderId)"
                        Button("Resolve") {
                            onResolve(row)
                        }
                        .buttonStyle(.borderedProminent)
                        .disabled(busyKey == key)
                    }
                }
            }
        }
    }
}

File: pegasusX/apps/supplier-app-android/app/src/main/java/com/pegasusx/supplier/ui/screens/exceptions/ExceptionsList.kt
package com.pegasusx.supplier.ui.screens.exceptions

import androidx.compose.ui.res.stringResource

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.pegasusx.supplier.data.model.SupplierExceptionRow
import com.pegasusx.supplier.ui.theme.PegasusSpacing

private val resolvableKinds = setOf("CASH_DISCREPANCY", "CREDIT_NOTE_DRAFT", "CREDIT_FREEZE")

@Composable
fun ExceptionsList(
    rows: List<SupplierExceptionRow>,
    busyKey: String? = null,
    onResolve: (SupplierExceptionRow) -> Unit = {},
) {
    LazyColumn(
        modifier = Modifier.fillMaxSize(),
        contentPadding = PaddingValues(PegasusSpacing.lg),
        verticalArrangement = Arrangement.spacedBy(PegasusSpacing.sm),
    ) {
        items(rows, key = { it.orderId + it.kind }) { row ->
            ElevatedCard(modifier = Modifier.fillMaxWidth()) {
                Column(
                    Modifier.padding(PegasusSpacing.lg),
                    verticalArrangement = Arrangement.spacedBy(4.dp),
                ) {
                    Text(row.orderId, style = MaterialTheme.typography.titleMedium)
                    Text(stringResource(R.string.mobile_supplier_ui_kind_status, row.kind, row.status), style = MaterialTheme.typography.bodyMedium)
                    row.note?.let { Text(it, style = MaterialTheme.typography.bodySmall) }
                    row.manifestId?.let {
                        Text(stringResource(R.string.mobile_supplier_ui_manifest_it, it), style = MaterialTheme.typography.bodySmall)
                    }
                    if (row.kind.uppercase() in resolvableKinds) {
                        val key = "${row.kind}:${row.orderId}"
                        Button(
                            onClick = { onResolve(row) },
                            enabled = busyKey != key,
                            modifier = Modifier.padding(top = PegasusSpacing.xs),
                        ) {
                            Text("Resolve")
                        }
                    }
                }
            }
        }
    }
}
